ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -177,15 +177,28 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			rayJobInstance.Status.DashboardURL = clientURL
 		}
 
+		if rayJobInstance.Spec.SubmissionMode == rayv1.NoneMode {
+			logger.Info("SubmissionMode is NoneMode and the RayCluster is created. Transition the status from `Initializing` to `Waiting`.")
+			rayJobInstance.Status.JobDeploymentStatus = rayv1.JobDeploymentStatusWaiting
+			break
+		}
+
 		if rayJobInstance.Spec.SubmissionMode == rayv1.K8sJobMode {
 			if err := r.createK8sJobIfNeed(ctx, rayJobInstance, rayClusterInstance); err != nil {
 				return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, err
 			}
 		}
 
-		logger.Info("Both RayCluster and the submitter K8s Job are created. Transition the status from `Initializing` to `Running`.",
+		logger.Info("SubmissionMode is K8sJobMode and both RayCluster and the submitter K8s Job are created. Transition the status from `Initializing` to `Running`.",
 			"RayJob", rayJobInstance.Name, "RayCluster", rayJobInstance.Status.RayClusterName)
 		rayJobInstance.Status.JobDeploymentStatus = rayv1.JobDeploymentStatusRunning
+	case rayv1.JobDeploymentStatusWaiting:
+		rayJobId, found := rayJobInstance.ObjectMeta.Annotations[utils.RayJobSubmissionIdLabelKey]
```

Comment:
1. Check whether `SubmissionMode` is `NoneMode`. If not, return an error.
2. Add a log message

Reply:
I only return an error. Should the log message be added? I think it may be redundant.

Reply:
print `rayJobId` instead of RayJob's name

Reply:
Why to print `rayJobId`? [Here](https://github.com/ray-project/kuberay/pull/2364/files#diff-f10777dd022e918221782f3ea3f88c70293404f9e62ded13cbad66b68c6d19efR196-R198) if the `SubmissionMode` is `NoneMode` and it will return immediately without getting `rayJobId` from annotations.

Reply:
If the annotation is not set for some reason, the RayJob will remain in `JobDeploymentStatusWaiting` indefinitely. It's better to inform users that it is stuck in this status because the annotation is not set correctly.

It's OK if you prefer not to add the log because `SubmissionMode: NoneMode` should not be set by users directly.

Reply:
Okay. I've added a log.